test/hotspot/jtreg/runtime/NMT/SummarySanityCheck.java

```diff
@@ -64,8 +64,8 @@ public static void main(String args[]) throws Exception {
     long totalCommitted = 0, totalReserved = 0;
     long totalCommittedSum = 0, totalReservedSum = 0;
 
-    // Match '- <mtType> (reserved=<reserved>KB, committed=<committed>KB)
-    Pattern mtTypePattern = Pattern.compile("-\\s+(?<typename>[\\w\\s]+)\\(reserved=(?<reserved>\\d+)KB,\\scommitted=(?<committed>\\d+)KB\\)");
+    // Match '- <mtType> (reserved=<reserved>KB, committed=<committed>KB) and some times readonly=<readonly>KB
+    Pattern mtTypePattern = Pattern.compile("-\\s+(?<typename>[\\w\\s]+)\\(reserved=(?<reserved>\\d+)KB,\\scommitted=(?<committed>\\d+)KB((,\\sreadonly=(?<readonly>\\d+)KB)|)\\)");
```

Comment:
You can simplify this, since the only line where we want to see readonly is "Shared class space". That makes the test also more strict, since we do not want to see readonly on any other line.